OMCompiler/SimulationRuntime/c/simulation/solver/linearSystem.c

```diff
@@ -396,8 +396,10 @@ int freeLinearSystems(DATA *data, threadData_t *threadData)
 int solve_linear_system(DATA *data, threadData_t *threadData, int sysNumber, double* aux_x)
 {
   TRACE_PUSH
-  int success;
   int retVal;
+#pragma omp critical
+{
+  int success;
```

Comment:
This solves the problems regarding garbage collector and systems with many linear systems. It's not prity but in this PR I want only to parallelize the Jacobian evaluation inside DASSL and IDA.